Flaky test - k8sclusterreceiver - TestFactory

Seen here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/4689582202?check_suite_focus=true

```
=== RUN   TestFactory
E0103 09:50:52.233080   45629 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 47 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x23bbc40, 0x3840740})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:74 +0xe6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000017898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:48 +0xb0
panic({0x23bbc40, 0x3840740})
	/opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c41c0, 0xc0001722d0, 0xc00047e060, 0xc000017b00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c41c0, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0002d66d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000253580, {0x2912940, 0xc00010eeb0}, 0x1, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c41c0, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:71 +0xdf
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference

goroutine 47 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000017898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:55 +0x145
panic({0x23bbc40, 0x3840740})
	/opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c41c0, 0xc0001722d0, 0xc00047e060, 0xc000017b00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c41c0, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0002d66d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000253580, {0x2912940, 0xc00010eeb0}, 0x1, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c41c0, 0xc00047e060)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:71 +0xdf
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver	0.111s
```

This was seen right in a commit that touched this component: https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/d202af20407be117499b86939924a85fb957d6db

@secustor, as the author of that PR (#6436), would you be able to take a look and see if that caused this failure?

Another failure:

```
=== RUN   TestFactory
E0104 16:33:26.125806   45654 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 46 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x23bbc40, 0x3840740})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:74 +0xe6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00015b898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:48 +0xb0
panic({0x23bbc40, 0x3840740})
	/opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c22a0, 0xc0000f0480, 0xc00012e000, 0xc00015bb00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c22a0, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0001536d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00014a100, {0x2912940, 0xc000112370}, 0x1, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c22a0, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:71 +0xdf
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference

goroutine 46 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00015b898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/runtime/runtime.go:55 +0x145
panic({0x23bbc40, 0x3840740})
	/opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c22a0, 0xc0000f0480, 0xc00012e000, 0xc00015bb00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c22a0, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0001536d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00014a100, {0x2912940, 0xc000112370}, 0x1, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c22a0, 0xc00012e000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.1/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.1/pkg/util/wait/wait.go:71 +0xdf
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver	0.113s
```

Seen here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/4704428527?check_suite_focus=true

--- a/internal/k8sconfig/config.go
+++ b/internal/k8sconfig/config.go
@@ -21,10 +21,16 @@ import (
 	"os"
 
 	quotaclientset "github.com/openshift/client-go/quota/clientset/versioned"
+	k8sruntime "k8s.io/apimachinery/pkg/util/runtime"
 	k8s "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
+
+func init() {
+	k8sruntime.ReallyCrash = false
+	k8sruntime.PanicHandlers = []func(interface{}){}
+}
 
 // AuthType describes the type of authentication to use for the K8s API
 type AuthType string

--- a/internal/k8sconfig/go.mod
+++ b/internal/k8sconfig/go.mod
@@ -4,6 +4,7 @@ go 1.17
 
 require (
 	github.com/openshift/client-go v0.0.0-20210521082421-73d9475a9142
+	k8s.io/apimachinery v0.23.5
 	k8s.io/client-go v0.23.5
 )
 
@@ -34,7 +35,6 @@ require (
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 	k8s.io/api v0.23.5 // indirect
-	k8s.io/apimachinery v0.23.5 // indirect
 	k8s.io/klog/v2 v2.30.0 // indirect
 	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
 	k8s.io/utils v0.0.0-20211116205334-6203023598ed // indirect

